src/llaminator.ts

```diff
@@ -52,45 +65,26 @@ export class Llaminator {
         'fileselected', Llaminator.prototype.onFileSelected.bind(this));
     this.elements.container.addEventListener(
         'itemdeleted', Llaminator.prototype.onItemDeleted.bind(this));
-  }
-
-  /**
-   * Clears the existing content from the container.
-   */
-  clearContainer() {
-    const { container } = this.elements;
 
-    while (container.firstChild) {
-      container.firstChild.remove();
+    switch (layout) {
+      default:
+        this.layout = new LlamaVerticalScrollLayout();
+        break;
     }
   }
 
   /**
-   * Populates all stored items in the container. Right now this works by removing all items from
-   * the container, and then repopulating it with everything known in Llaminator. This will be
-   * revisited when we decide on ranking, lazy updating, and item removal.
+   * Clears the existing content from the container, then renders |this.layout| in it.
    */
-  async populate() {
+  async resetContainer() {
     const { container } = this.elements;
 
-    const database = await this.storage;
-    const items = [];
-
-    for (const item of await database.list()) {
-      const blob = await database.getFile(item.id);
-      if (!blob) {
-        // TODO: Display an error, or prune this item from the database.
-        continue;
-      }
-
-      const url = URL.createObjectURL(blob);
-
-      items.push(html`
-          <llama-item .storage=${database} id=${item.id} src=${url}>
-          </llama-item>`);
+    while (container.firstChild) {
+      container.firstChild.remove();
     }
 
-    render(items, container);
+    render(this.layout, this.elements.container);
+    this.layout.refresh(await this.storage);
```

Comment:
Should we refresh before re-render()ing? That avoids a layout. (Suppose `getFile` is super slow...)

Reply:
done